### DISCUSSION TITLE
Set custom audio quality

### DISCUSSION THREAD

**Discussion by openaspace** [jamulussoftware/jamulus#3112] [General]
Hello.
I have a private server and the audio quality is poor.. how I can set waw 24bit  stereo?

Thank you.

**Comment by pljones** [comment]
Have you tried Sonobus?

**Reply by openaspace** [reply]
I use It everyday..
But there is the difference latency problem with each participant.
Placing a central server with 5ms it's better

**Reply by pljones** [reply]
Currently the OPUS codec doesn't offer lossless compression at anything other than 16bit 48KHz, as far as I know.

If you know of a backwards compatible codec that can be dropped into Jamulus, we'd be please to see a demonstration.

**Reply by openaspace** [reply]
hello, I see a lot of platform offering high quality (like wav 16/24bit) i'm not an expert of "free" open source terms...
but what i listen is that with sonobus the quality is high, why don't have the same in jamulus, where the sys admin can enable server side? we are a trio with 1gbps upload bandwidth.. 

"let's gas the engines on the track!"

**Reply by dcorson-ticino-com** [reply]
Jamulus was designed at a time when bandwidth was a precious good.  
Everything is compressed, but the best quality setting is still very good.
I have tried adding a non-compressed PCM quality, but gave up. It is a major deal.
I would encourage you to try your hand at adding a non-compressed quality if the present quality isn't good enough for you.

For me Jamulus' biggest quality is that it never crashes and just works reliably for hours at a time.
My attempts at using JamKazam and SonoBus have all ended after frustrating sessions full of serial player dropout and crashing.

**Reply by openaspace** [reply]
Hello.
Yes I already use high quality in jamulus but the depth audio quality of sonobus remain Unparalleled.. 

The drops of sonobus and other appliances are only relative to the bad server location or p2p connections between participants.

My experience is that if the server is really near the users, the speed and latency to the main country internet backbones is really less instead of the p2p direct connections... 

Indeed .. the latency of my users with central server is lower even if the distance is higher .. using an Amazon ec2 vps in Milan.

Better performance in p2p only with users with the same ISp in fiber when in the same " virtual ISp fiber LAN "

**Reply by dcorson-ticino-com** [reply]
>> The drops of sonobus and other appliances are only relative to the bad server location or p2p connections between participants.

I do not think this is true.
If the same 5 people can jump from sonobus with constant dropouts to Jamulus with no problems at all, the problem is not the connections.
IMHO the protocol used by sonobus and JamKazam is not stable.
As I often play in groups across the Atlantic using p2p to reduce the latency would be very interesting.
But we have tried those programs in all possible variations.  They are simply not stable.
Jamulus, however, is rock solid.  It just works.

**Reply by openaspace** [reply]
My experience with live television links in SRT technology is the same about latency, central server on Aws with low latency that works as hub produce less latency from A to B (central server) to C instead of A to C.

Values are like A to C 25ms.. instead of A to B (5ms) B to C (7ms) plus 4ms of CPU server processing=16vs25

**Reply by ann0see** [reply]
Research in the past has shown that jitter is the main problem. The question is: what is the problem you experience: Clicks, Pops, dropouts? Or not enough frequencies?
The former is a problem with unreliable internet transfer while the second one could be improved with higher Bitrate/no compression

**Reply by openaspace** [reply]
I'm asking here about better Audio in jamulus.. there is no problem with the audio network.

No network problems with jamulus no network problem with sonobus where I send at the same time 24 audio channels in 24bit and the difficulty is only relative to the different and more high ping for each participant in p2p instead of the central server... And therefore my question about jamulus is for obtaining lossless quality audio.

**Reply by pljones** [reply]
You've had a solution to that offered to you: provide an OPUS-compatible, lossless codec that seamlessly integrates with Jamulus.  Work has started on this a number of times and never borne fuit, unfortunately.
